### ISSUE TITLE
Tailwind CSS class names are removed `hover:`

### ISSUE THREAD

**Issue by docapotamus** [a-h/templ#162]
Apologies if I've missed something in the documentation but I'm having some issues using TailwindCSS with templ.

```
<a href={templ.URL(href)} class={
    templ.KV("bg-stone-800 text-white", active),
    templ.KV("text-stone-400", !active),
    "hover:text-white hover:bg-stone-800 group flex gap-x-3 rounded-md p-2 text-sm leading-6 font-semibold"}>
```

This renders out to:

```
bg-stone-800 text-white --templ-css-class-safe-name group flex gap-x-3 rounded-md p-2 text-sm leading-6 font-semibold
```

I've tried multiple ways of handling this, `templ.SafeCSS` doesn't help as it makes the enitre string `--templ-css-class-unknown-type"`.

If I don't use a CSS expression just `class="name hover:name name"` everything works fine. Potentially I'm missing some way of making this method dynamic.

Basically I just want to keep `hover:` in the class name. Is there a better way of acheiving this?

P.S. Thanks very much for templ. It's definetley the nicest Templating engine I've worked with in Go.

**Comment by a-h** [comment]
Sorry about the issue, it looks like it was an oversight during the addition of optional CSS class rendering functionality. The tests all had CSS class names that didn't trigger the CSS safety features, hence missing it.

I've done a PR, does that look like what you were expecting?

For background, CSS class names are sanitised by default, since templ can't tell whether they came from a user controlled variable (unsafe), or a programmer controlled constant (safe).

It looks like you spotted that you have to mark Go variables that you've set yourself (i.e. programmer controlled) as being safe by converting them to a `templ.ConstantCSSClass` type using the `templ.SafeClass` function.

But... I missed the code to handle that specific KV generic type.

Thanks for the great bug report and the nice words! 😁

**Comment by docapotamus** [comment]
@a-h thanks for quick response and the PR. It looks good to me, I'll check out that version in the code base later this morning and report back.

**Comment by docapotamus** [comment]
@a-h glad the issue with using `SafeClass` in `KV` is resolved. Looking at the PR, I realised that I should have been using `SafeClass` rather than `SafeCSS` in the string element and that resolved my issue in that case too.
